wandb/apis/public/api.py

```diff
@@ -680,14 +680,22 @@ def from_path(self, path):
         )
 
     def _parse_project_path(self, path):
-        """Return project and entity for project specified by path."""
+        """Return project, entity, and organization for project specified by path."""
```

Comment:
curious here though - is it necessary for this function to return the org name too? The project path (as the name suggests) should only need entity/project. 

won't it just make sense to return the orgEntity, projectName if the project is a registry project and avoid returning the org name altogether?

Reply:
I see your angle and there is probably a better way to refactor things. The reason why I explicitly make organization outputted here is that its hard to determine what the entity is set to. The entity can either be what the user settings are or what the user inputted as the path. 
In the case for a registry if a user inputs `wandb-registry-...` then the entity = `settings_entity` and we don't want to be accidentally set as the the org-entity.

An alternative is we would have to do this every time we call `_parse_project_path` or `_parse_artifact_path`  we also do something along the lines of this: 
```
entity, project = _parse_project_path(self, path)
if is_registry_project(project):
  parts = path.split("/")
  org = parts[0] if len(parts) == 2 else ""
```
I feel like thats just a lot of places to make this change and it might as well go into the `_parse....` functions